### PR TITLE
suppress some warnings in generated opensplice code

### PR DESCRIPTION
**Pull request by wjwwood**
This is similar to the `-Wno-` type arguments passed to the compiler for `rmw_connext` generated code. These warnings are out of our control, since it comes from code generated by OpenSplice.

Connects to ros2/ros2#53

### CODE REVIEW

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake [hunk 1]
```diff
@@ -77,6 +77,24 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
   endif()
 endforeach()
 
+# If not on Windows, disable some warnings with Connext's generated code
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: Comment referencing Connext.

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake [hunk 2]
```diff
@@ -77,6 +77,24 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
...
+if(NOT WIN32)
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(_opensplice_compile_flags)
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(_opensplice_compile_flags
+      "-Wno-unused-but-set-variable "
+    )
+  endif()
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: The variable should be set in all cases.

**Comment by esteve** [comment]
+1

File: rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake [hunk 3]
```diff
@@ -77,6 +77,24 @@ foreach(_idl_file ${rosidl_generate_interfaces_IDL_FILES})
   endif()
 endforeach()
 
+# If not on Windows, disable some warnings with Connext's generated code
+if(NOT WIN32)
+  if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
+    set(_opensplice_compile_flags)
+  elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+    set(_opensplice_compile_flags
+      "-Wno-unused-but-set-variable "
```
Comment (dirk-thomas) [inline rosidl_typesupport_opensplice_cpp/cmake/rosidl_typesupport_opensplice_cpp_generate_interfaces.cmake]: Same as https://github.com/ros2/rmw_connext/pull/40/files#r31542673

**Comment by dirk-thomas** [comment]
Same comment as https://github.com/ros2/rmw_connext/pull/40#issuecomment-108120545

**Pull request merged**